### DISCUSSION TITLE
Check for encryption type on auth traffic?

### DISCUSSION THREAD

**Discussion by RollingMoses** [Security-Onion-Solutions/securityonion#6202] [Unsupported Versions] [answered]
Hi, there! I would like to check for legacy, insecure authentication and ciphers. For example, we have a task to make sure RC4 is not used in our environment. I was checking through Zeek, but I haven't found something related to that, yet. If you all have used Security Onion to reveal that kind traffic and have some tips to share, I'd appreciate it!

Distributed deployment
Version 2.3.80
No host-based logging such as sysmon, Wazuh, etc. Just passive network taps.

**Comment by dougburks** [comment, marked as answer]
https://docs.zeek.org/en/master/scripts/policy/protocols/ssl/weak-keys.zeek.html
https://docs.securityonion.net/en/latest/zeek.html#configuration

**Reply by RollingMoses** [reply]
Thanks, @dougburks ! Looks like that's giving me a lot of interesting data now. For the benefit of others and to burn in my new knowledge, here's what I did:

I made the connection that there are scripts listed in https://docs.securityonion.net/en/latest/zeek.html#configuration that cross reference the scripts I saw on the left on https://docs.zeek.org/en/master/scripts/policy/protocols/ssl/weak-keys.zeek.html. So, I took the @load section of the /opt/so/saltstack/default/pillar/zeek/init.sls and added that to the end of the global.sls file, adding in the "policy/protocols/ssl/weak-keys" config:

zeek:
  local:
    '@load':
      - misc/loaded-scripts
      - tuning/defaults
      - misc/capture-loss
      - misc/stats
      - frameworks/software/vulnerable
      - frameworks/software/version-changes
      - protocols/ftp/software
      - protocols/smtp/software
      - protocols/ssh/software
      - protocols/http/software
      - protocols/dns/detect-external-names
      - protocols/ftp/detect
      - protocols/conn/known-hosts
      - protocols/conn/known-services
      - protocols/ssl/known-certs
      - protocols/ssl/validate-certs
      - protocols/ssl/log-hostcerts-only
      - protocols/ssl/weak-keys     ##   <-- This is the line item added
      - protocols/ssh/geo-data
      - protocols/ssh/interesting-hostnames
      - protocols/http/detect-sqli
      - frameworks/files/hash-all-files
      - frameworks/files/detect-MHR
      - policy/frameworks/notice/extend-email/hostnames
      - ja3
      - hassh
      - intel
      - cve-2020-0601
      - securityonion/bpfconf
      - securityonion/communityid
      - securityonion/file-extraction

I restarted Zeek to force the update on the sensors and I quickly started seeing weak keys notices. I switched to Kibana and filtered for "notice.note.security: SSL::Weak_Cipher" and saw notices about RC4 use.
